src/App.tsx

```diff
@@ -1,17 +1,149 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classnames from 'classnames';
 import './App.css';
 
-import users from './api/users';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { FullTodo } from './react-app-env';
+
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const fullTodo: FullTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [selectUserId, setSelectUserId] = useState(0);
+  const [todos, setTodos] = useState(fullTodo);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setUserIdError] = useState(false);
+  const [correctString, setCorrectString] = useState(true);
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (!/^[a-zA-Zа-яА-Я\d\s]+$/.test(event.target.value)) {
+      event.preventDefault();
+      setCorrectString(false);
```

Comment:
Will be better to have only one variable in state that should be responsible on the title errors